src/main/java/com/minecolonies/entity/EntityCitizen.java

```diff
@@ -624,6 +624,10 @@ public void onLivingUpdate()
             pickupItems();
             cleanupChatMessages();
             updateColonyServer();
+            if(!(this.getDesiredActivity() == EntityCitizen.DesiredActivity.SLEEP))
+            {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use the opposite operator ("!=") instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1940)